lib/puppet/provider/nameservice.rb

```diff
@@ -89,11 +89,19 @@ def section
       end
     end
 
-    def validate(name, value)
+    # This awful hack of passing `myself` is brought to you by the
+    # fact that this method isn't marked private, so I can't move it
+    # to be an instance method like I'd rather do - BER
+    def validate(name, value, myself=nil)
```

Comment:
Would it be terrible to copy it as an instance method, with a note to remove this for Puppet 5?